src/rnr_score_hv/innov_netcdf.py

```diff
@@ -0,0 +1,329 @@
+"""
+Copyright 2022 NOAA
+All rights reserved.
+
+Collection of methods to facilitate file/object retrieval
+
+"""
+from collections import namedtuple
+from dataclasses import dataclass, field
+from datetime import datetime
+import netCDF4
+
+from rnr_score_hv.config_base import ConfigInterface
+from rnr_score_hv import file_utils
+
+valid_metrics = [
+    'temperature',
+    'spechumid',
+    'uvwind'
+]
+
+VALID_STATS = [
+    'bias',
+    'count',
+    'rmsd'
+]
+
+
+@dataclass
+class Region:
+    ''' region object storing region name and min/max latitude bounds '''
+    name: str
+    min_lat: float
+    max_lat: float
+
+    def __post_init__(self):
+        if self.min_lat > self.max_lat:
+            msg = f'min_lat must be less than max_lat - min_lat: {self.min_lat}, '\
+                f'max_lat: {self.max_lat}'
+            raise ValueError(msg)
+        if self.max_lat < self.max_lat:
+            msg = f'min_lat must be greater than min_lat - min_lat: {self.min_lat}, '\
+                f'max_lat: {self.max_lat}'
+            raise ValueError(msg)
+
+        if abs(self.min_lat) > 90 or abs(self.max_lat) > 90:
+            msg = f'min_lat or max_lat is out of allowed range, must be greater' \
+                f' than -90 and let than 90 - min_lat: {self.min_lat}, ' \
+                f'max_lat: {self.max_lat}'
+            raise ValueError(msg)
+
+DEFAULT_REGIONS = [
+    Region('equatorial', -5, 5),
+    Region('global', -90, 90),
+    Region('north_hemis', 20, 60),
+    Region('tropics', -20, 20),
+    Region('south_hemis', -60, -20)
+]
+
+
+@dataclass
+class MetricMeta:
+    """
+    Information pertaining to where a particular measurement type
+    file can be found.
+
+    Parameters:
+    -----------
+    name: str - metric name (temperature, uvwind, or spechumid)
+    file_meta: dict - filename meta data containing file path
+               filename format, and cycle time
+    filename: str - configured full filename and absolute path
+    cycletime: datetime - for reference, the relevant cycle time
+               for the measurements
+    """
+    name: str
+    file_meta: dict
+    filename: str = field(default_factory=str, init=False)
+    cycletime: datetime = field(init=False)
+
+    def __post_init__(self):
+        try:
+            # build filename from the config meta data
+            file_path = self.file_meta.get('filepath')
+            cycle = self.file_meta.get('cycle')
+
+            self.cycletime = datetime.strptime(
+                cycle,
+                self.file_meta.get('cycletime_str')
+            )
+
+            filename_str = self.file_meta.get('filename_str')
+
+            new_fn = filename_str.replace('metric', self.name)
+
+            self.filename = file_path
+            self.filename += '/' + self.cycletime.strftime(new_fn)
```

Comment:
I feel conflicted here. Sounds like we are enforcing that the netcdf file always has a cycle time in its name. Seems like we are pigeonholing ourselves. The harvester will work just fine if the name is completely random.

Reply:
This harvester was designed specifically for the UFS-RNR outputs, but I could modify it such that it only checks the cycletime if it's configured.

Reply:
in your yamls in future PRs i saw a template for how cycletime is applied to the file name. I liked that. means you dont have to have cycle time in the filename. but i dont see the template been used here.

Reply:
What if we leave it as it for now.  We can always simply remove that handling and make the code that creates the config dict create the full path.  Then all we would need to do is check file existence/permissions/size validation.

Reply:
i am fine with that.